pkg/kubectl/cmd/cp.go

```diff
@@ -162,13 +162,13 @@ func checkDestinationIsDir(dest fileSpec, f cmdutil.Factory, cmd *cobra.Command)
 }
 
 func copyToPod(f cmdutil.Factory, cmd *cobra.Command, stdout, stderr io.Writer, src, dest fileSpec) error {
-	if len(src.File) == 0 {
+	if len(src.File) == 0 || len(dest.File) == 0 {
 		return errFileCannotBeEmpty
 	}
 	reader, writer := io.Pipe()
 
 	// strip trailing slash (if any)
-	if strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/") {
+	if strings.HasSuffix(string(dest.File[:len(dest.File)-1]), "/") {
```

Comment:
It seems we only care about omitting this check if `dest.File` is an empty string, so why not keep what you had previously, but check to see if len(dest.File) > 0

Reply:
Do you mean like this:
`if len(dest.File) > 0 && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/")`
Actually we have checked if dest.File is empty at first place of this function.
I think what we should check here is  if dest.File ends with slash. If so, strip it. However, when `dest.File == "/"`, we should not strip it. This cause the bug below:
kubectl cp /tmp/test   pod1:/
It will fail because dest.File ends up with empty string as said above.
So , maybe we can do like this:
`if dest.File != "/" && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/")`

Reply:
> Actually we have checked if dest.File is empty at first place in this function.

Ah thanks, didn't see your change above.

> I think what we need to check here is whether dest.File ends with slash. If so, strip it. However, when dest.File == "/", 
> we should not strip it. This cause the bug below:
> kubectl cp /tmp/test pod1:/
> It will fail because dest.File ends up with empty string as said above.
> So , maybe we can do like this:
> if dest.File != "/" && strings.HasSuffix(string(dest.File[len(dest.File)-1]), "/")

Makes sense to me, thanks